packages/api-rx/src/index.ts

```diff
@@ -0,0 +1,297 @@
+import assert from '@polkadot/util/assert';
+import isUndefined from '@polkadot/util/is/undefined';
+import logger from '@polkadot/util/logger';
+
+import SubmittableExtrinsic from './SubmittableExtrinsic';
+import { StorageFunction } from '@polkadot/types/StorageKey';
+
+const l = logger('api-rx');
+
+const INIT_ERROR = `Api needs to be initialised before using, listen on 'whenReady'`;
+
+export default class ApiRx extends E3.EventEmitter implements RxApiInterface {
+  private _extrinsics?: SubmittableExtrinsics;
+  private _genesisHash?: Hash;
+  private _storage?: QueryableStorage;
+  private _rpc: RpcRx;
+  private _runtimeMetadata?: RuntimeMetadata;
+  private _runtimeVersion?: RuntimeVersion;
+
+  // Observable that returns the first time we are connected and loaded
+  readonly isReady: Observable<ApiRx>;
+
+  /**
+   * @description Creates an ApiRx instance using the supplied provider. Returns an Observable containing the actual Api instance.
+   * @param wsProvider Optional WebSocket provider that is passed to the class contructor
+   * @example
+   * <BR>
+   *
+   * ```javascript
+   * import Api from '@polkadot/api-rx';
+   *
+   * Api.create().subscribe((api) => {
+   *   api.st.timestamp.now((timestamp) => {
+   *     console.log(`lastest block timestamp ${timestamp}`);
+   *   });
+   * });
+   * ```
+   */
+  static create (wsProvider?: WsProvider): Observable<ApiRx> {
+    return new ApiRx(wsProvider).isReady;
+  }
+
+  /**
+   * @param wsProvider An optional WebSocket provider from rpc-provider/ws. If not specified, it will default to connecting to the localhost with the default port
+   * @example
+   * <BR>
+   *
+   * ```javascript
+   * import Api from '@polkadot/api-rx';
+   *
+   * new Api().isReady.subscribe((api) => {
+   *   api.rpc.newHead().subscribe((header) => {
+   *     console.log(`new block #${header.blockNumber.toNumber()}`);
+   *   });
+   * });
+   * ```
+   */
+  constructor (wsProvider?: WsProvider) {
+    super();
+
+    this._rpc = new RpcRx(wsProvider);
+
+    this.isReady = from(this.init());
+  }
+
+  /**
+   * @description Observable that carries the connected state for the provider. Results in a boolean flag that is true/false based on the connectivity.
+   */
+  get isConnected (): Observable<boolean> {
+    return this.rpc.isConnected();
+  }
+
+  /**
+   * @description Contains the genesis Hash of the attached chain. Apart from being useful to determine the actual chain, it can also be used to sign immortal transactions.
+   */
+  get genesisHash (): Hash {
+    assert(!isUndefined(this._genesisHash), INIT_ERROR);
+
+    return this._genesisHash as Hash;
+  }
+
+  /**
+   * @description Contains all the raw rpc sections and their subsequent methods in the API as defined by the jsonrpc interface definitions.
+   * @example
+   * <BR>
+   *
+   * ```javascript
+   * api.rpc.chain
+   *   .newHead()
+   *   .subscribe((header) => {
+   *     console.log('new header', header);
+   *   });
```

Comment:
could shorten to `.subscribe((header) => console.log('new header', header));`

Reply:
Explicitly done that way to be as readable as possible.